### PR TITLE
Update URLs in scheduler.md with correct folder path

### PR DESCRIPTION
**Pull request by 2014BDuck**
**Which issue(s) this PR fixes**:

The `generic_scheduler.go` is moved under `scheduler` folder, not `scheduler/core` folder. This commit updated the description & the link to the code file.

Note that `legacy_registry.go` is removed as well in `master` branch. We may need to fix the link and description as well after release.

### CODE REVIEW

File: contributors/devel/sig-scheduling/scheduler.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ We are dividing scheduler into three layers from high level:
   This is the main() entry that does initialization before calling the scheduler framework.
 - [pkg/scheduler/scheduler.go](http://releases.k8s.io/HEAD/pkg/scheduler/scheduler.go):
   This is the scheduler framework that handles stuff (e.g. binding) beyond the scheduling algorithm.
-- [pkg/scheduler/core/generic_scheduler.go](http://releases.k8s.io/HEAD/pkg/scheduler/core/generic_scheduler.go):
+- [pkg/scheduler/generic_scheduler.go](http://releases.k8s.io/HEAD/pkg/scheduler/generic_scheduler.go):
```
Comment (alculquicondor) [inline contributors/devel/sig-scheduling/scheduler.md:14]: Actually, this should be pkg/scheduler/framework/runtime

(this document is very old)

**Review by alculquicondor — COMMENTED** [review]
Actually, it might be time to remove this file.
Is there anything here not already covered in https://github.com/kubernetes/community/blob/503990668645701b3484482729c1c7ec4665b448/contributors/devel/sig-scheduling/scheduling_code_hierarchy_overview.md ?

**Comment by 2014BDuck** [comment]
> Actually, it might be time to remove this file. Is there anything here not already covered in scheduling_code_hierarchy_overview.md 

Hi. I am new to Kubernetes scheduler. So those doc do help and I think we should correct those mistake anyway since they are linked in the [README](https://github.com/kubernetes/community/blob/master/contributors/devel/README.md) for developers.

So we could either fix the `scheduler.md`, or change the description in `README.md` above (and maybe point to `scheduling_code_hierarchy_overview.md`?).

And we have the same folder structure in `scheduling_code_hierarchy_overview.md` tho (which is old), I would like to change it a little bit as well.

**Comment by alculquicondor** [comment]
I recommend:
1. remove `scheduler.md` after moving anything that is not covered to `scheduling_code_hierarchy_overview.md`
2.  rename `scheduling_code_hierarchy_overview.md`  back to `scheduler.md`

**Comment by Huang-Wei** [comment]
> 1. remove `scheduler.md` after moving anything that is not covered to `scheduling_code_hierarchy_overview.md`

It seems the only part is "scheduler extensibility". The current portion is way too dated, and we need a new doc (scheduler_framework_plugins.md can be referred).

>  2. rename `scheduling_code_hierarchy_overview.md`  back to `scheduler.md`

Another option is to remove scheduler.md, and list all well-maintained documents in [devel/README.md#SIG Scheduling](https://github.com/kubernetes/community/blob/master/contributors/devel/README.md#sig-scheduling). For now it only lists 3 docs - we have more such as scheduler_framework_plugins.md, scheduler_queues.md, etc. So we can polish the part in devel/README.md like this:

- Understanding the Kubernetes Scheduler scheduling_code_hierarchy_overview.md
- Scheduler Algorithm in Kubernetes scheduler_algorithm.md
- Understanding how Pods are queued in Kubernetes Scheduler scheduler_queues.md
- Extening Kubernetes Scheduler TBD
- Scheduler Benchmarking scheduler_benchmarking.md

**Comment by 2014BDuck** [comment]
OK understanding it's not a small change if we want to have updated docs referring to the current scheduler structure.

I've modified the [devel/README.md#SIG Scheduling](https://github.com/kubernetes/community/blob/master/contributors/devel/README.md#sig-scheduling):
- replaced `scheduler.md` with  `scheduling_code_hierarchy_overview.md` 
- linked the extension guide to `scheduler_framework_plugins.md `
- linked `scheduler_queues.md` too

And the **"scheduler extensibility"** part in `scheduler.md` is moved to `scheduling_code_hierarchy_overview.md` as well. I feel like it's already out of date but could not update it with the correct content, which obviously requires more experience.

Would like to modify them to the latest version after reading more scheduler codes. (probably a few weeks later?

File: contributors/devel/sig-scheduling/scheduling_code_hierarchy_overview.md [hunk 2]
```diff
@@ -66,6 +66,22 @@ Consists of the following four steps ran in the same order:
 In case of processing of any of the extension points fails, `Unreserve` operation
 of all `Reserve` plugins is invoked (e.g. free resources allocated for a gang of pods).
 
+## Scheduler extensibility
+
+The scheduler is extensible: the cluster administrator can choose which of the pre-defined scheduling policies to apply, and can add new ones.
+
+### Modifying policies
```
Comment (alculquicondor) [inline contributors/devel/sig-scheduling/scheduling_code_hierarchy_overview.md:73]: This is actually gone now. I would say just remove it for now. The new API is mostly documented in the website:

https://kubernetes.io/docs/reference/scheduling/config/
https://kubernetes.io/docs/reference/config-api/kube-scheduler-config.v1beta2/

And the implementation details of these are already briefly mentioned in `Initial startup configuration` below

File: contributors/devel/sig-scheduling/scheduling_code_hierarchy_overview.md [hunk 3]
```diff
@@ -66,6 +66,22 @@ Consists of the following four steps ran in the same order:
 In case of processing of any of the extension points fails, `Unreserve` operation
 of all `Reserve` plugins is invoked (e.g. free resources allocated for a gang of pods).
 
+## Scheduler extensibility
```
Comment (Huang-Wei) [inline contributors/devel/sig-scheduling/scheduling_code_hierarchy_overview.md:69]: This deserves a standalone document. I can compose one later. For now, just remove this section (as well as "Modifying policies") to keep this document unchanged.

File: contributors/devel/README.md [hunk 4]
```diff
@@ -117,10 +117,14 @@ See the [kubernetes/release](https://github.com/kubernetes/release) repository f
...
 ### SIG Scheduling
 
-* **Understanding the Kubernetes Scheduler** [scheduler.md](sig-scheduling/scheduler.md)
+* **Understanding the Kubernetes Scheduler** [scheduling_code_hierarchy_overview.md](sig-scheduling/scheduling_code_hierarchy_overview.md)
 
 * **Scheduler Algorithm in Kubernetes** [scheduler_algorithm.md](sig-scheduling/scheduler_algorithm.md)
 
+* **Understanding how Pods are queued in Kubernetes Scheduler** [scheduler_queues.md](sig-scheduling/scheduler_queues.md)
+
+* **Extending with scheduler framework plugins** [scheduler_framework_plugins.md](sig-scheduling/scheduler_framework_plugins.md)
```
Comment (Huang-Wei) [inline contributors/devel/README.md:126]: Plugins are part of scheduler's extensibility, not all of it. You can remove this bullet for now. We will come up with a dedicated document talking about scheduler's overall extensibility, and then link back here.

**Comment by 2014BDuck** [comment]
OK so we changed the devel `README.md` only in this PR, which linked `scheduling_code_hierarchy_overview.md`, and added `scheduler_queues.md`. Please review again. Many Thanks!

I would like to contribute some docs later if possible. Current docs do help but still hard for newbies to learn how to debug the scheduler :P

**Comment by alculquicondor** [comment]
/lgtm

Please do contribute. We will be happy to review.

**Review by cblecker — APPROVED** [review]

**Pull request merged**